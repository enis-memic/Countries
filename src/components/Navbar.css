.navigation {
  background-color: rgba(114, 37, 93, 0.518);
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 30px;
}

.heading {
  font-size: 40px;
  color: white;
  text-decoration: none;
  font-family: 'Lato', sans-serif;
  font-weight: 700;
}

.position {
  display: flex;
  gap: 20px;
  cursor: pointer;
  font-size: 20px;
}

.icon-nav {
  color: white;
  width: 5rem;
}

@media (max-width: 480px) {
  .navigation {
    gap: 25px;
    display: flex;
    align-items: center;
  }

  .icon-nav {
    color: white;
    width: 2rem;
  }
}
