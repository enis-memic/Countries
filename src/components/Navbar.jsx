import React from 'react';
import { NavLink } from 'react-router-dom';
import { FaMicrophone, FaEllipsisH } from 'react-icons/fa';
import { SlArrowLeft } from 'react-icons/sl';
import './Navbar.css';

function Navbar() {
  return (
    <div className="navigation">
      <NavLink to="/" element>
        <SlArrowLeft className="icon-nav" />
      </NavLink>
      <h2 className="heading">Countries Statistics</h2>

      <div className="position">
        <FaMicrophone className="icon-nav" />
        <FaEllipsisH className="icon-nav" />
      </div>
    </div>
  );
}

export default Navbar;
