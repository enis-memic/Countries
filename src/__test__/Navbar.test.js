import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { store } from '../redux/store';
import Navbar from '../components/Navbar';

describe('Navbar rendering', () => {
  test('Navbar rendering', () => {
    const render = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>
    );
    expect(render).toMatchSnapshot();
  });
});
