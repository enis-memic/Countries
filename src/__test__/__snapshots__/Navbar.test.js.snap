// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar rendering Navbar rendering 1`] = `
<div
  className="navigation"
>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <svg
      className="icon-nav"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
      />
    </svg>
  </a>
  <h1
    className="heading"
  >
    Countries
  </h1>
  <div
    className="position"
  >
    <svg
      className="icon-nav"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 352 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
      />
    </svg>
    <svg
      className="icon-nav"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
      />
    </svg>
  </div>
</div>
`;
